kong/tools/utils.lua

```diff
@@ -297,24 +319,7 @@ do
     for _, key in ipairs(keys) do
       local value = args[key]
       if type(value) == "table" then
-        for sub_key, sub_value in pairs(value) do
-          if no_array_indexes then
-            query[#query+1] = encode_args_value(key, sub_value, raw)
-
-          else
-            if type(sub_key) == "number" then
-              query[#query+1] = encode_args_value(("%s[%s]")
-                                  :format(key, tostring(sub_key)), sub_value,
-                                          raw)
-
-            else
-              query[#query+1] = encode_args_value(("%s.%s")
-                                  :format(key, tostring(sub_key)), sub_value,
-                                          raw)
-
-            end
-          end
-        end
+        recursive_encode_args(key, value, raw, no_array_indexes, query)
```

Comment:
At the top of `encode_args()` is a comment documenting its behavior and differences with `ngx.encode_args()`. We should add a mention for such nested array parameters now.